src/DataTypes/Serializations/SerializationDecimalBase.cpp

```diff
@@ -61,6 +72,16 @@ void SerializationDecimalBase<T>::deserializeBinaryBulk(IColumn & column, ReadBu
     size_t initial_size = x.size();
     x.resize(initial_size + limit);
     size_t size = istr.readBig(reinterpret_cast<char*>(&x[initial_size]), sizeof(FieldType) * limit);
+    if constexpr (std::endian::native == std::endian::big && sizeof(T) >= 2)
+    {
+        for (size_t i = 0; i < limit; i++)
+        {
+            char *start = reinterpret_cast<char*>(&x[initial_size + i]);
+            char *end = start + sizeof(FieldType);
+            std::reverse(start, end);
+        }
```

Comment:
You should be able to use `transformEndianness` here as well.

Reply:
Tried, but somehow the templates don't resolve for decimal types. so let's keep it as it is.